packages/adapter-drizzle/src/mysql/index.ts

```diff
@@ -0,0 +1,174 @@
+import type { DbClient, Schema } from "./schema"
+import { and, eq } from "drizzle-orm"
+import type { Adapter } from "next-auth/adapters"
+import { v4 as uuid } from "uuid"
+
+export function MySqlAdapter(
+  client: DbClient,
+  { users, sessions, verificationTokens, accounts }: Schema
+): Adapter {
+  return {
+    createUser: async (data) => {
+      const id = uuid()
+
+      await client.insert(users).values({ ...data, id })
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, id))
+        .then((res) => res[0])
+    },
+    getUser: async (data) => {
+      return (
+        client
+          .select()
+          .from(users)
+          .where(eq(users.id, data))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    getUserByEmail: async (data) => {
+      return (
+        client
+          .select()
+          .from(users)
+          .where(eq(users.email, data))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    createSession: async (data) => {
+      await client.insert(sessions).values(data)
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then((res) => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return (
+        client
+          .select({
+            session: sessions,
+            user: users,
+          })
+          .from(sessions)
+          .where(eq(sessions.sessionToken, data))
+          .innerJoin(users, eq(users.id, sessions.userId))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      await client.update(users).set(data).where(eq(users.id, data.id))
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, data.id))
+        .then((res) => res[0])
+    },
+    updateSession: async (data) => {
+      await client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then((res) => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      await client
+        .insert(accounts)
+        .values(rawAccount)
+        .then((res) => res[0])
+    },
+    getUserByAccount: async (account) => {
```

Comment:
Hello! Thank you for your hard work on this. 

I've been testing this out in a project I'm using next auth and drizzle for. I could be wrong here but I think this method will return the wrong user sometimes.

It appears to be selecting every user, performing a join, then returning the first user which will always be the first user registered in the database.

For example, I register an email account under 'test@example.com', log out and then register through an oauth provider that has a different email as the primary email (so a separate account is made), then log out and log back in through that same oauth provider you should be logged in as the first user in the database, not your oauth user. 

It's a bit of a convoluted example but it's the only one I've tested lol

Again thank you for your hard work so far!

Reply:
Good point. This should:
1) query `accounts` table w/ `provider` and `providerAccountId`
2) If exists, query `users` by `account.id` to get user

Reply:
Good catch! Thank you!

Reply:
Yes, updated again!